crates/api/src/site/block.rs

```diff
@@ -19,6 +19,11 @@ pub async fn block_instance(
 ) -> Result<Json<BlockInstanceResponse>, LemmyError> {
   let instance_id = data.instance_id;
   let person_id = local_user_view.person.id;
+  let local_site = SiteView::read_local(&mut context.pool()).await?;
```

Comment:
It looks like you already have the `instance_id` , its on `local_user_view.person.instance_id`

Reply:
Ah perfect!